### DISCUSSION TITLE
My renewed involvement in pyfakefs

### DISCUSSION THREAD

**Discussion by davidlbaird** [jmcgeheeiv/pyfakefs#819] [General]
I think I owe an explanation to the pyfakefs community for my renewed interest in pyfakefs. Many years ago, when pyfakefs was pushed from Google's internal code base to code.google.com, the intent was to make this unit test tool freely available, and updating changes developed at Google to the PyPI community. When code.google.com was shut down, it became very clear that there were branches of pyfakefs and it made more sense to me to allow others, notably John McGehee, to merge the different branches and continue maintenance. I am very grateful to all who have contributed to pyfakefs over the years.

At Google, we continued to maintain our own internal branch, and a few years ago, saw the need to port our internal uses to the github pyfakefs, especially so that we could make use of important PyPI libraries that test their code with pyfakefs. So my own internal initiative to deprecate our old branch began. Over the past few months, I led an internal project to port all of our test case uses to the latest pyfakefs... I expect to delete our old branch this month. So, I'd say that pyfakefs has always been very important for Google, and we have many use cases, and now, it is also very important for us to be involved in the pyfakefs implementation on github.

There were some features that were lost over the recent years, and although I have found workarounds, I would like to reintroduce one or two. However, in general, the fs fixture will become our recommended in-house interface for POSIX, Windows and MacOS file systems, but it is not the only use of pyfakefs. I would like to explain how to make use of the fake filesystem for unique situations that are also commonly in use at Google, and not (yet) available externally. So, I think there is a lot for Google to contribute to pyfakefs.

So, that should sum up my re-emergence after many years of being absent.

Thanks,
David

**Comment by mrbean-bremen** [comment]
We are very happy about that new (or renewed) involvement (I think I can say this for @jmcgeheeiv too). As pyfakefs has originated at Google, I had always wondered if it was still used there, or just abandoned after making it public on GitHub. I like that Google will use the `fs` fixture in the future, and I'm looking forward to your contributions for making the package better usable at Google and elsewhere.

**Comment by jmcgeheeiv** [comment]
Thank you for the explanation, and welcome back.

I was wondering what Google's relationship with pyfakefs was all these
years.
